src/modules/QueryBuilder.spec.ts

```diff
@@ -400,9 +431,40 @@ describe('QueryBuilder', () => {
         })
         expect(filter).toBeTruthy()
         expect(
-          (filter!['foobar'] as MappedFilter)[ComparisonQueryOperatorEnum.GREATER_THAN]
+          (filter!['foobar'] as MappedFilter)[
+            ComparisonQueryOperatorEnum.GREATER_THAN
+          ]
         ).toEqual(2)
       })
+
+      it('should return the correct value ISODate with the specified date. <YYYY-mm-dd>', () => {
```

Comment:
Here (and in many other places) the test names are suboptimal. Right now you added multiple tests that have the exact same name, but do different things - and at the same time the test names don't really describe what they are testing. 
Improving it can be as simple as adding "gt filter should return..." to the test name.